src/funcs.c

```diff
@@ -57,6 +57,36 @@
 
 #include <src/hookintrprtr.h>
 
+static ModuleStateHandle StateHandle;
+typedef struct {
+    Int RecursionDepth;
+    Obj ExecState;
+} ModuleState;
+
+void IncRecursionDepth(void)
+{
+    MODULE_STATE(RecursionDepth)++;
```

Comment:
I really like how this cleanly decouples the state of this module from the global `GAPState`. For nitpicks on the actual implementation, see elsewhere in this review :-).